pkg/controlplane/reconcilers/endpointsadapter.go

```diff
@@ -123,6 +123,7 @@ func (adapter *EndpointsAdapter) EnsureEndpointSliceFromEndpoints(namespace stri
 func endpointSliceFromEndpoints(endpoints *corev1.Endpoints) *discovery.EndpointSlice {
 	endpointSlice := &discovery.EndpointSlice{}
 	endpointSlice.Name = endpoints.Name
+	endpointSlice.Namespace = endpoints.Namespace
```

Comment:
@liggitt this is the only non-test change in order to keep the test files code equal and not diverge, otherwise I need to modify the helpers_test.go and probably something on the tests, they are failing without this line because the semantic comparison fails without the namespace.
This is not an issue when using the real client since it's adds the namespace

Reply:
yeah, this lgtm